lnwire/open_channel2.go

```diff
@@ -0,0 +1,309 @@
+	// should use this key and their per-commitment point to derive the
+	// revocation key for the commitment transaction.
+	RevocationPoint *btcec.PublicKey
+
+	// PaymentPoint is the base payment point for the sending party. This
+	// key should be combined with the per commitment point for a
+	// particular commitment state in order to create the key that should
+	// be used in any output that pays directly to the sending party, and
+	// also within the HTLC covenant transactions.
+	PaymentPoint *btcec.PublicKey
+
+	// DelayedPaymentPoint is the delay point for the sending party. This
+	// key should be combined with the per commitment point to derive the
+	// keys that are used in outputs of the sender's commitment transaction
+	// where they claim funds.
+	DelayedPaymentPoint *btcec.PublicKey
+
+	// HtlcPoint is the base point used to derive the set of keys for this
+	// party that will be used within the HTLC public key scripts. This
+	// value is combined with the receiver's revocation base point in order
+	// to derive the keys that are used within HTLC scripts.
+	HtlcPoint *btcec.PublicKey
+
+	// FirstCommitmentPoint is the first commitment point for the sending
+	// party. This value should be combined with the receiver's revocation
+	// base point in order to derive the revocation keys that are placed
+	// within the commitment transaction of the sender.
+	FirstCommitmentPoint *btcec.PublicKey
+
+	// ChannelFlags is a bit-field which allows the initiator of the
+	// channel to specify further behavior surrounding the channel.
+	// Currently, the least significant bit of this bit field indicates the
+	// initiator of the channel wishes to advertise this channel publicly.
+	ChannelFlags FundingFlag
+
+	// UpfrontShutdownScript is the script to which the channel funds should
+	// be paid when mutually closing the channel. This field is optional,
+	// and has a length prefix, so a zero will be written if it is not set
+	// and its length followed by the script will be written if it is set.
+	UpfrontShutdownScript DeliveryAddress
+
+	// ChannelType is the explicit channel type the initiator wishes to
+	// open.
+	ChannelType *ChannelType
+
+	// LeaseExpiry represents the absolute expiration height of a channel
+	// lease. This is a custom TLV record that will only apply when a leased
+	// channel is being opened using the script enforced lease commitment
+	// type.
+	LeaseExpiry *LeaseExpiry
+
+	// ExtraData is the set of data that was appended to this message to
+	// fill out the full maximum transport message size. These fields can
+	// be used to specify optional data such as custom TLV fields.
+	//
+	// NOTE: Since the upfront shutdown script MUST be present (though can
+	// be zero-length) if any TLV data is available, the script will be
+	// extracted and removed from this blob when decoding. ExtraData will
+	// contain all TLV records _except_ the DeliveryAddress record in that
```

Comment:
I think it can also be removed in the ReadElement method or ? [lnwire.go](https://github.com/lightningnetwork/lnd/blob/359a5638f5a18918b0bec8597acf54a50e287eec/lnwire/lnwire.go#L441)

Reply:
I don't think there's really a reason to remove the upfront shutdown script from `ExtraBytes`.  We don't do it for any other TLVs.

So I just removed the outdated comment.

Reply:
Ok not sure whether I am on the wrong track, but I though that we could delete the serialization of the deliveryaddress in the lnwire package to disincentivizing writing or using it and putting it onto the wire. But maybe I am misunderstanding something here. Imo this does not mean deleting something from the ExtraBytes or ?

I am talking about the two following code parts:

ReadElement: https://github.com/lightningnetwork/lnd/blob/359a5638f5a18918b0bec8597acf54a50e287eec/lnwire/lnwire.go#L912

WriteElement:
https://github.com/lightningnetwork/lnd/blob/359a5638f5a18918b0bec8597acf54a50e287eec/lnwire/lnwire.go#L441

Reply:
Looks like we still need that code for the `shutdown` message.

Reply:
It seems to me that for the delivery address the bytes encoding method is used to write it to the wire.
https://github.com/lightningnetwork/lnd/blob/d263a01a2d3bb3570538418a326a8ed165e500a8/lnwire/typed_delivery_addr.go#L39

Reply:
ohh thank you, you are right.